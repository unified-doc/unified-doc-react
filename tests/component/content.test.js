import React from 'react';
import renderer from 'react-test-renderer';

import DocComponent from '../../lib/component';
import {
  markdownContent,
  htmlContent,
  jsContent,
  jsonContent,
} from '../fixtures';

const sanitizeSchema = {
  attributes: {
    '*': ['className', 'style'],
    mark: ['dataMarkId', 'id'],
  },
  clobberPrefix: '',
};

describe('content', () => {
  describe('html content', () => {
    it('renders html content (.html extension)', () => {
      const options = {
        content: htmlContent,
        filename: 'doc.html',
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders code block (.any extension)', () => {
      const options = {
        content: htmlContent,
        filename: 'doc.any',
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders html content (.md extension)', () => {
      const options = {
        content: htmlContent,
        filename: 'doc.md',
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('markdown content', () => {
    it('renders markdown content (.html extension)', () => {
      const options = {
        content: markdownContent,
        filename: 'doc.html',
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders code block (.any extension)', () => {
      const options = {
        content: markdownContent,
        filename: 'doc.any',
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders markdown content (.md extension)', () => {
      const options = {
        content: markdownContent,
        filename: 'doc.md',
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('code block content (sanitized)', () => {
    it('renders JS content to code block (.js extension)', () => {
      const options = {
        content: jsContent,
        filename: 'doc.js',
        sanitizeSchema,
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders JSON content to code block (.js extension)', () => {
      const options = {
        content: jsonContent,
        filename: 'doc.json',
        sanitizeSchema,
      };
      const tree = renderer.create(<DocComponent options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
