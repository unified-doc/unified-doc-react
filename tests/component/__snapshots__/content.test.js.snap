// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content code block content (sanitized) renders JS content to code block (.js extension) 1`] = `
<div>
  <div>
    <pre
      className="language-js"
    >
      <code
        className="language-js"
      >
        function hello() {
  return "world";
}
      </code>
    </pre>
  </div>
</div>
`;

exports[`content code block content (sanitized) renders JSON content to code block (.js extension) 1`] = `
<div>
  <div>
    <pre
      className="language-json"
    >
      <code
        className="language-json"
      >
        {"one":2,"three":[true,false,null,"four",5],"six":{"7":"eight","nine":0}}
      </code>
    </pre>
  </div>
</div>
`;

exports[`content html content renders code block (.any extension) 1`] = `
<div>
  <div>
    <pre>
      <code>
        &lt;blockquote&gt;&lt;strong&gt;some&lt;/strong&gt;
content&lt;/blockquote&gt;
      </code>
    </pre>
  </div>
</div>
`;

exports[`content html content renders html content (.html extension) 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        some
      </strong>
      
content
    </blockquote>
  </div>
</div>
`;

exports[`content html content renders html content (.md extension) 1`] = `
<div>
  <div />
</div>
`;

exports[`content markdown content renders code block (.any extension) 1`] = `
<div>
  <div>
    <pre>
      <code>
        &gt; **some** markdown content
      </code>
    </pre>
  </div>
</div>
`;

exports[`content markdown content renders markdown content (.html extension) 1`] = `
<div>
  <div>
    &gt; **some** markdown content
  </div>
</div>
`;

exports[`content markdown content renders markdown content (.md extension) 1`] = `
<div>
  <div>
    <blockquote>
      

      <p>
        <strong>
          some
        </strong>
         markdown content
      </p>
      

    </blockquote>
  </div>
</div>
`;
